src/irr/asset/CGLILoader.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+MIT License
+Copyright (c) 2019 AnastaZIuk
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "CBufferLoaderBIN.h"
+
+//#ifdef _IRR_COMPILE_WITH_GLI_
+#include "gli/gli.hpp"
+//#else
+//	#error "Need GLI"
+//#endif
+
+namespace irr
+{
+	namespace asset
+	{
+		asset::SAssetBundle CBufferLoaderBIN::loadAsset(io::IReadFile* _file, const asset::IAssetLoader::SAssetLoadParams& _params, asset::IAssetLoader::IAssetLoaderOverride* _override, uint32_t _hierarchyLevel)
+		{
+			if (!_file)
+				return {};
+
+			SContext ctx(_file->getSize());
+			ctx.file = _file;
+
+			const char* filename = _file->getFileName().c_str();
+			gli::texture texture = gli::load(filename);
+			if (texture.empty())
+			{
+				os::Printer::log("Failed to load texture at path ", filename, ELL_ERROR);
+				return {};
+			}
+			
+		    gli::gl GL(gli::gl::PROFILE_GL33);
```

Comment:
whats the difference between this and PROFILE_KTX

Reply:
I don't know actually and I can't find anything about it, so I have created an issue https://github.com/g-truc/gli/issues/172

I hope they will clarify us it

Reply:
good.

Reply:
btw you opened the issue on a fork of the original repo